crates/cxx-qt-lib/src/core/qanystringview.cpp

```diff
@@ -0,0 +1,46 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#include "cxx-qt-lib/qanystringview.h"
+
+#include "../assertion_utils.h"
+
+// QAnyStringView has two members.
+// A union of (void*, char*, char_16*) and a size_t.
+// https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/text/qanystringview.h
+assert_alignment_and_size(QAnyStringView,
+                          alignof(::std::size_t),
+                          sizeof(::std::size_t) + sizeof(void*));
+
+static_assert(::std::is_trivially_copy_assignable<QAnyStringView>::value);
+static_assert(::std::is_trivially_copy_constructible<QAnyStringView>::value);
+
+static_assert(::std::is_trivially_destructible<QAnyStringView>::value);
+
+static_assert(QTypeInfo<QAnyStringView>::isRelocatable);
+
+namespace rust {
+namespace cxxqtlib1 {
+
+QAnyStringView
+qanystringviewInitFromRustString(::rust::Str string)
+{
+  // Note that rust::Str here is borrowed
+  // and we convert back from UTF-8 to UTF-16
+  return QAnyStringView(string.data(), string.size());
+}
+
+::rust::isize
+qanystringviewLen(const QAnyStringView& string)
+{
+  // In Qt 5 the type was int now it is qsizetype, so we need to ensure the type
```

Comment:
guess this comment isn't required if it's only Qt 6?

Reply:
Correct, it was a copy & paste error from QString. Removed